src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,159 @@
+        DiagnosticDescriptors.DoNotUseHostConfigureServices.Id,
+        DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id
+     });
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            string id = diagnostic.Id;
+            switch (id)
+            {
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id:
+                    context.RegisterCodeFix(
+                    CodeAction.Create("Fix references to Logging properties on WebApplicationBuilder",
+                    cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, "Logging"),
+                    equivalenceKey: DiagnosticDescriptors.DoNotUseHostConfigureLogging.Id),
+                    diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DoNotUseHostConfigureServices.Id:
+                    context.RegisterCodeFix(
+                    CodeAction.Create("Fix references to Services properties on WebApplicationBuilder",
+                    cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, "Services"),
+                    equivalenceKey: DiagnosticDescriptors.DoNotUseHostConfigureServices.Id),
+                    diagnostic);
+                    break;
+
+                case string when id == DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id:
+                    context.RegisterCodeFix(
+                    CodeAction.Create("Fix references to Services properties on WebApplicationBuilder",
+                    cancellationToken => FixWebApplicationBuilder(diagnostic, context.Document, cancellationToken, "Configuration"),
+                    equivalenceKey: DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id),
+                    diagnostic);
+                    break;
+            }
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> FixWebApplicationBuilder(Diagnostic diagnostic, Document document, CancellationToken cancellationToken, string IDENTIFIER_METHOD_NAME)
+    {
+        DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root == null)
+        {
+            return document;
+        }
+
+        var diagnosticTarget = root.FindNode(diagnostic.Location.SourceSpan); //whole invocation starting from "builder"
+
+        if (diagnosticTarget is InvocationExpressionSyntax { } originalInvocation)
+        {
+            if (originalInvocation.Expression is not MemberAccessExpressionSyntax hostBasedInvocationMethodExpr
```

Comment:
It might help to assign this condition to a well-named variable to help that explains what this is checking for.